File: 22 - Comparing Numbers/Golang/main.go
package main

import (
	"fmt"
	"sort"
)

func highestinarr(arr []int) int {
	h := arr[0] // maxsize is hardcoded, so checking not needed here imo

	for _, v := range arr {
		if v > h {
			h = v
		}
	}
	return h
}

func hasduplicates(arr []int) bool {
	sort.Ints(arr)
	for i := 0; i < len(arr)-1; i++ {
		if arr[i] == arr[i+1] {
			return true
		}
	}
	return false
}

func main() {
	const maxsize = 3
	var numbers [maxsize]int
	for i := 0; i < maxsize; i++ {
		fmt.Print("Enter a number: ")
		fmt.Scanf("%d", &numbers[i])
	}

	if hasduplicates(numbers[:]) {
		fmt.Println("There are duplicates in this list.")
		return
	} else {
		fmt.Printf("The highest number is %d.", highestinarr(numbers[:]))
	}
}

File: 22 - Comparing Numbers/C++/main.cpp
#include <algorithm>
#include <iostream>

template <int Size> int highest_intarr(int (&arr)[Size]) {
    int h = arr[0]; // not checking because maxsize is a thing in main()
    for (int i : arr) {
        if (i > h)
            h = i;
    }
    return h;
}

template <int Size> bool hasduplicates_intarr(int (&arr)[Size]) {
    std::sort(arr, arr + Size);
    for (int i = 0; i < Size - 1; i++) {
        if (arr[i] == arr[i + 1])
            return true;
    }
    return false;
}

int main() {
    constexpr int maxsize = 3;
    int numbers[maxsize];

    for (int i = 0; i < maxsize; i++) {
        std::cout << "Enter a number: ";
        std::cin >> numbers[i];
    }

    if (hasduplicates_intarr(numbers)) {
        std::cout << "There are duplicates." << std::endl;
        return 0;
    } else {
        std::cout << "The largest number is " << highest_intarr(numbers)
                  << std::endl;
    }

    return 0;
}
